src/object/object_primitive.c

```diff
@@ -16399,21 +16399,24 @@ mr_get_compression_length (const char *string, int charlen)
  */
 
 int
-mr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
+pr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
 {
   char *compressed_string = NULL, *string = NULL, *str = NULL;
   int rc = NO_ERROR, str_length = 0, length = 0;
   lzo_uint compression_length = 0;
   lzo_bytep wrkmem = NULL;
   bool compressed = false;
-  int error_abort = 0;
+  int error_abort = -1;
 
   /* Checks to be sure that we have the correct input */
   assert (DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_VARNCHAR || DB_VALUE_DOMAIN_TYPE (value) == DB_TYPE_STRING);
   assert (DB_GET_STRING_SIZE (value) >= PRIM_MINIMUM_STRING_LENGTH_FOR_COMPRESSION);
 
-  error_abort = buf->error_abort;
-  buf->error_abort = 0;
+  if (buf->error_abort != NULL)
```

Comment:
I have checked again and every time we use the new `pr_get_size_and_write_string_to_buffer` function, the buffer should not be initialized, because that is the purpose of the function. So the `error_abort` check it's always useless. More over, every initialize is done with `OR_BUF_INIT`, which sets `error_abort` to `0`, so there will not be any `_setjmp` usage, therefore the `or_overflow(buf)` return is also not used. I am going to remove them both.